src/AutoRest.CSharp.V3/Output/Builders/SerializationBuilder.cs

```diff
@@ -150,24 +150,70 @@ private static XmlObjectSerialization BuildXmlObjectSerialization(ObjectSchema o
                 schemaTypeReference, elements.ToArray(), attributes.ToArray(), embeddedArrays.ToArray());
         }
 
+        private static IEnumerable<JsonPropertySerialization> GetPropertySerializationsFromBag(PropertyBag propertyBag)
+        {
+            foreach (Property property in propertyBag.Properties)
+            {
+                yield return new JsonPropertySerialization(
+                    property.SerializedName,
+                    property.CSharpName(),
+                    BuildSerialization(property.Schema, property.IsNullable()),
+                    false);
+            }
+
+            foreach ((string name, PropertyBag innerBag) in propertyBag.Bag)
+            {
+                JsonPropertySerialization[] serializationProperties = GetPropertySerializationsFromBag(innerBag).ToArray();
+                JsonObjectSerialization objectSerialization = new JsonObjectSerialization(new SchemaTypeReference(null, false), serializationProperties, null);
+                yield return new JsonPropertySerialization(name, name, objectSerialization, true);
+            }
+        }
+
         private static JsonObjectSerialization BuildJsonObjectSerialization(ObjectSchema objectSchema, TypeReference schemaTypeReference, bool isNullable)
         {
             List<JsonPropertySerialization> serializationProperties = new List<JsonPropertySerialization>();
             foreach (var schema in EnumerateHierarchy(objectSchema))
             {
-                foreach (Property property in schema.Properties!)
+                PropertyBag propertyBag = new PropertyBag();
+                propertyBag.Properties.AddRange(schema.Properties ?? Array.Empty<Property>());
+                PopulatePropertyBag(propertyBag, 0);
```

Comment:
Shouldn't this call be outside of the loop?

Reply:
`GetPropertySerializationsFromBag` uses a populated property bag. So, the next line needs that to add the correct property serializations to the list.

Reply:
We can pull next line out of the loop too.